src/mono/wasm/runtime/pthreads/worker/index.ts

```diff
@@ -0,0 +1,34 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+/// <reference lib="webworker" />
+
+import { makeChannelCreatedMonoMessage, pthread_ptr } from "../shared";
+
+export type ThreadCreatedCallback = (pthread_ptr: pthread_ptr, main_port: MessagePort) => void;
+
+const threadCreatedCallbacks = Array<ThreadCreatedCallback>();
```

Comment:
This collection is in each worker ?
Does it contain list of all other workers ?
What is it good for ?

What makes it `callback` ?

Reply:
This is a generic callback to do something every time that Emscripten creates a pthread inside of a worker - before the pthread gets attached to mono inside the runtime.

The list of callbacks is in dotnet.js, so each worker will have its own list (but in practice this is going to be the same list in every thread since they will just register callbacks just after the wasm module is loaded inside the worker.

Reply:
The later instances would not have others which registered earlier than they existed, right ?